.github/workflows/build-all.yaml

```diff
@@ -0,0 +1,42 @@
+# (C) Copyright Confidential Containers Contributors
+# # SPDX-License-Identifier: Apache-2.0
+#
+# Build binary with all providers.
+---
+name: build-all
+on: [pull_request]
+jobs:
+  build_job:
+    name: build
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        type:
+          - dynamic
+          - static
+        runner:
+          - ubuntu-latest
+        go_version:
+          - 1.18
+        include:
+          - type: dyamic
+            providers: "aws,azure,ibmcloud,vsphere,libvirt"
+          - type: static
+            providers: "aws,azure,ibmcloud,vsphere"
+
+    steps:
+      - name: Checkout the pull request code
+        uses: actions/checkout@v3
+      - name: Setup Golang version ${{ matrix.go_version }}
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go_version }}
+      - name: Install build dependencies
```

Comment:
nit: we can add a condition for only when libvirt is included (dynamic) only no?

Reply:
yes. Fixed in new update